### ISSUE TITLE
gitlab actions build broken Windows release (cannot move map with mouse)

### ISSUE THREAD

**Issue by zdechlak** [forkserf/forkserf#279]
for some reason I can't move the map with the mouse / Windows 10

**Comment by tlongstretch** [comment]
hmm, for comparison, does it work for you with Freeserf?  https://github.com/freeserf/freeserf/releases/tag/v0.3

I am wondering if this is simply a Windows10 issue relevant to both Freeserf & Forkserf, or if it is related to something changed in Forkserf only.

**Comment by zdechlak** [comment]
in 0.3 it works

**Comment by zdechlak** [comment]
i remember a few years ago, in the first versions this bug http://jonls.dk/freeserf/ - it was solved by Jon Lund Steffensen

**Comment by tlongstretch** [comment]
Thanks, I will do some testing to see if I can reproduce.  I have made a lot of changes to the SDL Mouse events in Forkserf relating to zooming and double/special/right-clicking

**Comment by zdechlak** [comment]
basically, I can only use the mouse to zoom and select a point on the map
nothing else works

**Comment by tlongstretch** [comment]
You are holding a button in to drag the map around, right?  It will not move the screen unless one of the mouse buttons is held in

**Comment by zdechlak** [comment]
I hold the mouse buttons - sometimes the map skips a bit, it does it by lagging - and it doesn't work properly / / the same bug was once in the first remake projects

**Comment by zdechlak** [comment]
https://user-images.githubusercontent.com/2237225/213540807-2fce44a6-ea25-4d7d-9a96-4041bf48194b.mp4

**Comment by tlongstretch** [comment]
Yes I agree that is not expected behavior, I do not see that issue on linux.  I will see if I can reproduce this on Windows7, and if not I'll look for a windows 10 machine to test with

**Comment by zdechlak** [comment]
make it a priority because it's hard to play with the arrow keys on the keyboard

**Comment by tlongstretch** [comment]
I am able to reproduce this in Windows7 so I should be able to fix easily, looking into it now

**Comment by tlongstretch** [comment]
the issue is something with the compilation.  If I use VSCode from Windows7 and build it the mouse scroll works fine, but the same code built by GitLab Workflow/Action does not work.  Looking to see if there is something I can adjust to make it build a working copy...

**Comment by tlongstretch** [comment]
it seems this is not related to Forkserf changes, the newest FreeSerf builds for windows have the same mouse issue.  I suspect they are defaulting to a newer SDL2 that breaks something.

I have updated the Forkserf v0.5.2 Release page to include the working Win64 version that I compiled and tested, try this: https://github.com/forkserf/forkserf/releases/download/v0.5.2/Forkserf-0.5.2-Win64.zip

**Comment by zdechlak** [comment]
hooray it works ❤

**Comment by zdechlak** [comment]
topic to be closed

**Comment by tlongstretch** [comment]
I want to keep this open as reminder to occasionally look into fixing the actual gitlab actions build
